clang/include/clang/APINotes/Types.h

```diff
@@ -749,6 +750,17 @@ struct Context {
 
   Context(ContextID id, ContextKind kind) : id(id), kind(kind) {}
 };
+
+/// A temporary reference to an Objective-C selector, suitable for
+/// referencing selector data on the stack.
+///
+/// Instances of this struct do not store references to any of the
+/// data they contain; it is up to the user to ensure that the data
+/// referenced by the identifier list persists.
+struct ObjCSelectorRef {
+  unsigned NumPieces;
```

Comment:
I don't understand this.  Why have `NumPieces` when `llvm::ArrayRef<T>::size` gives us that?

Reply:
Good point, we're not actually using `NumPieces` anywhere. Removed it.